clusterDeployer.py

```diff
@@ -284,22 +284,36 @@ def create_cluster(self) -> None:
         logger.info(cfg)
         self._ai.create_cluster(cluster_name, cfg)
 
-    def create_masters(self) -> None:
+    def ensure_infraenv_created(self, *, is_workers: bool) -> str:
```

Comment:
The intent is good, but adding a boolean flag to a function makes the code harder to read.
Now, you have to add `if-else`  things all over the place.

What if we instead pass in a list of nodes for which you are creating the infra env. commit ae3bcb99 added `self.masters_arch` and `self.wokers_arch`, but can be derived from a list of nodes. 

Also, this change doesn't below in this PR. It should be a separate refactoring PR.

Reply:
> The intent is good, but adding a boolean flag to a function makes the code harder to read.

The alternative you see before. It's "easy" to read one part of it. But an non-trivial block of code is duplicated in subtly different ways. Since you don't see the block side-by-side, it's almost impossible to see how they differ. That is not "easier".

I drop this patch.